cli/internal/cmd/upgradeapply.go

```diff
@@ -331,32 +359,38 @@ func (u *upgradeApplyCmd) upgradeAttestConfigIfDiff(cmd *cobra.Command, stableCl
 	if equal {
 		return nil
 	}
-
+	cmd.Println("The configured attestation config is different from the attestation config in the cluster.")
+	diffStr, err := diffAttestationCfg(clusterAttestationConfig, newConfig)
+	if err != nil {
+		return fmt.Errorf("diffing attestation configs: %w", err)
+	}
+	cmd.Println("The following changes will be applied to the attestation config:")
+	cmd.Println(diffStr)
 	if !flags.yes {
-		cmd.Println("The configured attestation config is different from the attestation config in the cluster.")
-		diffStr, err := diffAttestationCfg(clusterAttestationConfig, newConfig)
-		if err != nil {
-			return fmt.Errorf("diffing attestation configs: %w", err)
-		}
-		cmd.Println("The following changes will be applied to the attestation config:")
-		cmd.Println(diffStr)
 		ok, err := askToConfirm(cmd, "Are you sure you want to change your cluster's attestation config?")
 		if err != nil {
 			return fmt.Errorf("asking for confirmation: %w", err)
 		}
 		if !ok {
-			cmd.Println("Skipping upgrade.")
-			return nil
+			return errors.New("aborting upgrade since attestation config is different")
 		}
 	}
-	if err := u.upgrader.UpdateAttestationConfig(cmd.Context(), newConfig); err != nil {
-		return fmt.Errorf("updating attestation config: %w", err)
+	if err := u.upgrader.BackupConfigMap(cmd.Context(), constants.JoinConfigMap); err != nil {
```

Comment:
I would prefer if the backup is triggered somewhere else, more visibly. I realize this PR is close to being merged. I would be fine with a hint in the method's comment and possible TODO here.